superset/alfred/api.py

```diff
@@ -0,0 +1,105 @@
+from flask_babel import lazy_gettext as _
+from flask_login import current_user
+from marshmallow import ValidationError
+
+
+from superset.advanced_data_type.schemas import (
+    AdvancedDataTypeSchema,
+)
+from superset.alfred.schemas import RetainToAlfredSchema
+from superset.constants import RouteMethod
+from superset.extensions import event_logger, security_manager
+from superset.alfred.utils import retain_eml_to_alfred
+
+logger = logging.getLogger(__name__)
+
+config = app.config
+
+class AlfredRestApi(BaseApi):
+    """
+    Fission rest endpoint to proxy hogwarts fission
+    """
+
+    allow_browser_login = True
+    include_route_methods = {RouteMethod.POST}
+    resource_name = "alfred"
+    class_permission_name = "AdvancedDataType"
+
+    openapi_spec_tag = "Alfred"
+    openapi_spec_component_schemas = (AdvancedDataTypeSchema,)
+    
+    add_model_schema = RetainToAlfredSchema()
+
+
+    @protect()
+    @safe
+    @expose("/retain-eml-record", methods=["POST"])
+    @permission_name("read")
+    @event_logger.log_this_with_context(
+        action=lambda self, *args, **kwargs: f"{self.__class__.__name__}.post",
+        log_to_statsd=False,  # pylint: disable-arguments-renamed
+    )
+    def post(self, **kwargs: Any) -> Response:
+        """
+        Fetch the email metadata associated with the email IDs sent in the 
+        request and send the data to alfred to create a staged retention
+
+        Returns:
+            Response: 
+                200 Success, { result: "<alfred-url-for-staged-retention>"}
+        """
+        try:
+            request_payload = self.add_model_schema.load(request.json)
+        # This validates custom Schema with custom validations
+        except ValidationError as error:
+            return self.response_400(message=error.messages)
+
+        if not request_payload:
+            logger.info("no JSON payload in request")
+            return self.response_400('no JSON payload in request')
+        if 'email_ids' not in request_payload:
+            return self.response_400('No field email_ids found in json body of request')
+
+        logger.info("Payload is %s", request_payload)
+        email_ids = request_payload['email_ids']
+        dates = None
+        if 'dates' in request_payload:
+            dates = request_payload['dates']
+
+        alfred_env = os.environ.get("ALFRED_ENV")
+        if not alfred_env:
+            logger.info("ALFRED_ENV environment variable not set")
+            return self.response_400('ALFRED_ENV environment variable not set')
+        user = current_user
+        token = security_manager.get_on_behalf_of_access_token_with_cache(
+            user.username,
+            os.environ.get("SUPERSET_SCOPE"),
+            "superset",
+            cache_result=True,
+        )
+        status, result = retain_eml_to_alfred(email_ids, alfred_env, token, dates)
+        logger.info('result: %s', result)
```

Comment:
Do we want all these log messages at the info level? Just checking.

Reply:
I went back through and removed a bunch of unecessary logs and changed them to be error level logs where needed